src/service.js

```diff
@@ -37,7 +37,9 @@ class Service {
     const q = this.Model.find(query).lean(this.lean);
 
     // $select uses a specific find syntax, so it has to come first.
-    if (filters.$select && filters.$select.length) {
+    if (typeof filters.$select === 'string') {
```

Comment:
I think you might be able to just add this check to line 51, which would pass the string to the mongoose query.select()